internal/backend/local/backend_test.go

```diff
@@ -170,6 +283,75 @@ func TestLocal_addAndRemoveStates(t *testing.T) {
 	}
 }
 
+// Test logic that checks if two backends are going to conflict with which state
+// files they are managing.
+//
+// This test only covers comparison of local state backends that have no
+// non-default workspaces.
+func TestLocal_PathsConflictWith_defaultWorkspaceOnly(t *testing.T) {
+
+	// The base local backend that each test case is compared to
+	comparedLocal := Local{
+		StatePath:       DefaultStateFilename,
+		StateOutPath:    DefaultStateFilename,
+		StateBackupPath: DefaultStateFilename + DefaultBackupExtension,
+		// StateWorkspaceDir is not relevant in this test, as that only
+		// impacts non-default workspaces
+	}
+
+	conflicts := true
+	doesNotConflict := false
+
+	cases := map[string]struct {
+		comparedTo *Local
+		want       bool
+	}{
+		"a local state backend will conflict when compared to itself": {
+			comparedTo: &comparedLocal,
+			want:       conflicts,
+		},
+		"matching values for state path result in a conflict": {
+			comparedTo: &Local{
+				StatePath:    DefaultStateFilename, // conflicts
+				StateOutPath: "no/conflict.tfstate",
+			},
+			want: conflicts,
+		},
+		"a state path override is sufficient to stop conflict": {
+			comparedTo: &Local{
+				StatePath:         DefaultStateFilename, // would conflict if no override
+				StateOutPath:      DefaultStateFilename,
+				OverrideStatePath: "no-conflict.tfstate",
+			},
+			want: doesNotConflict,
+		},
+		"matching values for state out path does not get identified as conflict": {
+			comparedTo: &Local{
+				StatePath:    "no/conflict.tfstate",
+				StateOutPath: DefaultStateFilename, // matches but doesn't conflict
+			},
+			want: doesNotConflict,
+		},
```

Comment:
Does this make sense?

If two backends have the same StateOutPath then they're both writing to the same file. Is this not a conflict?

Reply:
Yes, does make sense. The -state-out flag isn't available for init commands, and this code under test is only relevant when migrating state during an init command.